File: LR2/27/27.swift
import Foundation // Импорт Foundation для работы с функциями ввода-вывода

func correctDate(day: Int, month: Int) -> Bool {
    return day >= 1 && day <= 31 && month >= 1 && month <= 12
}

func main() {
    // Чтение количества чисел
    guard let inputLine = readLine() else { return }
    let tokens = inputLine.components(separatedBy: .whitespaces)
    
    // Первый токен - количество чисел
    guard let count = Int(tokens[0]) else { return }
    
    // Обработка каждого числа
    for i in 1...count {
        guard i < tokens.count else { break }
        let token = tokens[i]
        
        // Проверка на корректность: все символы — цифры
        guard token.allSatisfy({ $0.isNumber }) else {
            print("NO", terminator: "")
            continue
        }
        
        var isCorrectDate = false
        
        if let number = Int(token) {
            if number >= 11 && number < 100 {
                let day = number / 10
                let month = number % 10
                isCorrectDate = correctDate(day: day, month: month)
            } else if number >= 100 && number < 1000 {
                var day = number / 10
                var month = number % 10
                isCorrectDate = correctDate(day: day, month: month)
                
                if !isCorrectDate {
                    day = number / 100
                    month = number % 100
                    isCorrectDate = correctDate(day: day, month: month)
                }
            } else if number >= 1000 && number <= 3112 {
                let day = number / 100
                let month = number % 100
                isCorrectDate = correctDate(day: day, month: month)
            }
        }
        
        print(isCorrectDate ? "YES" : "NO", terminator: "")
    }
}

main() // Запуск основной функции

File: LR2/27/27.kt
fun main() {
    // Читаем входные данные и разбиваем их на отдельные токены по пробелам
    val input = readLine()?.split(" ") ?: emptyList()
    
    // Первый элемент - количество чисел
    val n = input.firstOrNull()?.toIntOrNull() ?: 0
    
    // Обрабатываем каждое число, начиная со второго элемента
    input.drop(1).take(n).forEach { token ->
        processToken(token)
    }
}

fun processToken(token: String) {
    // Проверяем, что токен состоит только из цифр
    if (!token.matches(Regex("^\\d+$"))) {
        println("Invalid input")
        return
    }

    val number = token.toInt()
    var isCorrectDate = false

    when {
        number in 11..99 -> {
            val day = number / 10
            val month = number % 10
            isCorrectDate = correctDate(day, month)
        }
        number in 100..999 -> {
            // Проверяем два варианта разделения
            val day1 = number / 10
            val month1 = number % 10
            isCorrectDate = correctDate(day1, month1)
            
            if (!isCorrectDate) {
                val day2 = number / 100
                val month2 = number % 100
                isCorrectDate = correctDate(day2, month2)
            }
        }
        number in 1000..3112 -> {
            val day = number / 100
            val month = number % 100
            isCorrectDate = correctDate(day, month)
        }
    }

    println(if (isCorrectDate) "YES" else "NO")
}

fun correctDate(day: Int, month: Int): Boolean {
    return day in 1..31 && month in 1..12
}